src/test/java/de/unistuttgart/overworldbackend/MinigameTaskControllerTest.java

```diff
@@ -63,27 +69,42 @@ public void createBasicData() {
     final MinigameTask minigameTask1 = new MinigameTask();
     minigameTask1.setConfigurationId(UUID.randomUUID());
     minigameTask1.setGame("Bugfinder");
-    minigameTask1.setLocation("w1-p1");
+    minigameTask1.setIndex(1);
 
     final MinigameTask minigameTask2 = new MinigameTask();
     minigameTask2.setConfigurationId(UUID.randomUUID());
     minigameTask2.setGame("Moorhuhn");
-    minigameTask2.setLocation("w1-p2");
+    minigameTask2.setIndex(2);
+
+    final MinigameTask minigameTask3 = new MinigameTask();
+    minigameTask2.setConfigurationId(UUID.randomUUID());
+    minigameTask2.setGame("Crosswordpuzzle");
+    minigameTask2.setIndex(3);
+
+    final Dungeon dungeon = new Dungeon();
+    dungeon.setStaticName("Dungeon 1");
+    dungeon.setTopicName("Testtopic");
+    dungeon.setActive(true);
+    dungeon.setMinigameTasks(Set.of(minigameTask3));
+    dungeon.setNpcs(Set.of());
 
     final World world = new World();
     world.setStaticName("Winter Wonderland");
     world.setTopicName("UML Winter");
     world.setActive(true);
     world.setMinigameTasks(Set.of(minigameTask1, minigameTask2));
     world.setNpcs(Set.of());
-    world.setDungeons(Arrays.asList());
+    world.setDungeons(Arrays.asList(dungeon));
 
     final Lecture lecture = new Lecture("PSE", "Basic lecture of computer science students", Arrays.asList(world));
     initialLecture = lectureRepository.save(lecture);
 
     initialWorld = initialLecture.getWorlds().stream().findFirst().get();
     initialWorldDTO = worldMapper.worldToWorldDTO(initialWorld);
 
+    initialDungeon = initialLecture.getWorlds().stream().findFirst().get().getDungeons().stream().findFirst().get();
```

Comment:
```suggestion
    initialDungeon = initialLecture.getWorlds().parallelStream().map(World::getDungeons).flatMap(Collection::parallelStream).findAny().get();
```

Reply:
in my opinion this does not improve code